### DISCUSSION TITLE
aws distributed training struck !

### DISCUSSION THREAD

**Discussion by Davdi** [apache/incubator-mxnet#13644] [Q&A]
i use the command 
python  /home/ec2-user/incubator-mxnet/tools/launch.py -n 1 -s 1  -H hosts --sync-dst-dir /home/ec2-user/incubator-mxnet/ "source activate mxnet_p36 && python image_classification.py --model vgg11 --gpus 0 --kvstore dist_sync" 

and it stuck ,the info is this 

> 

018-12-14 02:33:36,036 INFO rsync /home/ec2-user/incubator-mxnet/example/gluon/ -> 172.31.38.3:/home/ec2-user/incubator-mxnet/
2018-12-14 02:33:36,036 INFO rsync /home/ec2-user/incubator-mxnet/example/gluon/ -> 172.31.35.130:/home/ec2-user/incubator-mxnet/
[02:33:37] src/van.cc:290: Bind to role=scheduler, id=1, ip=172.31.35.130, port=9093, is_recovery=0
INFO:root:Starting new image-classification task:, Namespace(batch_norm=False, batch_size=32, builtin_profiler=0, data_dir='', dataset='cifar10', dtype='float32', epochs=120, gpus='0', kvstore='dist_sync', log_interval=50, lr=0.1, lr_factor=0.1, lr_steps='30,60,90', mode=None, model='vgg11', momentum=0.9, num_workers=4, prefix='', profile=False, resume='', save_frequency=10, seed=123, start_epoch=0, use_pretrained=False, use_thumbnail=False, wd=0.0001)
[02:33:37] src/van.cc:398: ? => 1. Meta: request=0, timestamp=0, control={ cmd=ADD_NODE, node={ role=server, ip=172.31.35.130, port=46899, is_recovery=0 } }

> 

but if i delete the arg  --kvstore dist_sync ,just run 
python  /home/ec2-user/incubator-mxnet/tools/launch.py -n 1 -s 1  -H hosts --sync-dst-dir /home/ec2-user/incubator-mxnet/ "source activate mxnet_p36 && python image_classification.py --model vgg11 --gpus 0 " 
it run successfully

and the two instance are on the same Security groups
and rules 

All TCP | TCP | 0 - 65535 
SSH | TCP | 22 | 0.0.0.0/0

**Comment by vdantu** [comment]
@mxnet-label-bot add [question]

**Comment by Ishitori** [comment]
@Davdi, did you manage to find the issue? To me it seems that it should be connected to connectivity, because when you remove `--kvstore dist_sync` the default value is used, which for this tutorial is `device`. 

I also notice that you set n = 1, but mention "2 instances". Which configuration do you try to achieve? And what is the content of your hosts file?

**Comment by pinaraltinyilmaz** [comment]
@mxnet-label-bot add[Distributed]

**Comment by pinaraltinyilmaz** [comment]
@mxnet-label-bot add[distributed]

**Comment by pinaraltinyilmaz** [comment]
@mxnet-label-bot add [distributed]
